### PR TITLE
implemented different types of pagination

### PR DESCRIPTION
**Pull request by AshantiMutinta**
## PR summary
-Second iteration of pagination adds a new type which can paginate by time and limit
## testing/benchmarking notes

( if any manual testing or benchmarking was/should be done, add notes and/or screenshots here )

## followups

( any new tickets/concerns that were discovered or created during this work but aren't in scope for review here )

## changelog

- [ ] if this is a code change that effects some consumer (e.g. zome developers) of holochain core,  then it has been added to [our between-release changelog](https://github.com/holochain/holochain-rust/blob/develop/CHANGELOG-UNRELEASED.md) with the format 

```markdown
- summary of change [PR#1234](https://github.com/holochain/holochain-rust/pull/1234)
```

## documentation

- [x] this code has been documented according to our [docs checklist](https://hackmd.io/@freesig/Hk9AmKJNS)

### CODE REVIEW

File: app_spec/test/files/links.js [hunk 1]
```diff
@@ -1,4 +1,5 @@
 const { one, two } = require('../config')
+const sleep = require('sleep')
 
 module.exports = scenario => {
   scenario('delete_post', async (s, t) => {
```
Comment (zippy) [inline app_spec/test/files/links.js]: Why is this scenario called "delete_post"

File: doc/holochain_101/src/links/get_links.md [hunk 2]
```diff
@@ -10,9 +10,8 @@ Get Links allows the zome developer to query links from the DHT. The call accept
 `Timeout` : The timeout variable on the options specifies how long the query process should wait befor a response before it timesout
 `LinksStatusRequest` : This is a variable in which you can specify 3 modes, `All`,`Live`,`Delete`. This allows you to query the links based on crud_status in which `All` will return everything will `Live` will only return live links and `Delete` as such.
 `Headers`: boolean value which if set to true indicates that the link headers should also be returned.```
-`Pagination`: The pagination type has two variables which are page size and page number. These describe the page size which divides the links into pages and the page number
-which describes the page that we want to get.
-'Sort Order' : Allows get_links to define which order `Ascending` or `Descending` the links should be returned in
+`Pagination`: The pagination type has two variaants which are `Size` and `Time` These variants allow the user to paginate based on size parameters or time based parameters with size pagination and time pagination respectively.
```
Comment (zippy) [inline doc/holochain_101/src/links/get_links.md]:
```suggestion
`Pagination`: The pagination type has two variants which are `Size` and `Time` These variants allow the user to paginate based on choosing pages and specifying page size, or by specifying a point time and specifying a limit of the number of entries (based on the timestamp in the entry header) to return from that point.
```

File: app_spec/test/files/links.js (hunk 1 above)
Comment (AshantiMutinta) [reply app_spec/test/files/links.js]: Woops that is not a new scenario but I have just added sleep in there :)

File: app_spec/zomes/simple/code/src/lib.rs [hunk 3]
```diff
@@ -73,7 +73,16 @@ pub fn handle_get_my_links_with_tag(agent : Address,status_request:LinksStatusRe
...
 {
     let options = GetLinksOptions{
-        pagination : Some(Pagination{page_size : pagesize,page_number : pagenumber}),
+        pagination : Some(Pagination::Size(SizePagination{page_size : pagesize,page_number : pagenumber})),
+        ..GetLinksOptions::default()
+    };
+    hdk::get_links_with_options(&agent, LinkMatch::Exactly("authored_simple_posts"), LinkMatch::Any,options)
+}
+
+pub fn handle_get_my_links_with_time_pagination(agent : Address,from_seconds:i64,limit:usize) ->ZomeApiResult<GetLinksResult>
```
Comment (freesig) [inline app_spec/zomes/simple/code/src/lib.rs]:
```suggestion
pub fn handle_get_my_links_with_time_pagination(agent : Address, from_seconds: i64, limit: usize) -> ZomeApiResult<GetLinksResult>
```

Comment (freesig) [reply app_spec/zomes/simple/code/src/lib.rs]: Does cargo fmt need to be run? there's a few odd formatting things

**Pull request merged**